### PR TITLE
Ahmed/eng 3916 ios sdd generate and backup ed25519 key share 2

### PR DESCRIPTION
**Pull request by AhmedRagabIssa**
## Summary
add generate and backup ed25519 key share, adding button "Generate Solana and backup shares"

## Visuals
simulator_screenshot_5008D5B7-B887-42F6-BE02-6E9AA99C46AD

### Code
- Documentation updated: Pending
To be added later.

### Impact
- Breaking Changes: No
- Performance impact: No

### Testing
- Unit tests added: No
- E2E tests added: No

### Misc.
- Metrics, logs, or traces added: No

### CODE REVIEW

File: SPM Example/PortalSwift/ViewController.swift [hunk 1]
```diff
@@ -1614,23 +1708,24 @@ class ViewController: UIViewController, UITextFieldDelegate {
...
-                } catch {
-                    print("Failed to refresh the the ETH balance. \(error)")
-                }
-            }
+  private func startRefreshBalanceTimer() {
+    self.refreshBalanceTimer = Timer.scheduledTimer(withTimeInterval: 5.0, repeats: true) { [weak self] _ in
+      guard let self = self else { return }
+      Task {
+        do {
+          try await self.populateEthBalance()
```
Comment (Rshahatit) [inline SPM Example/PortalSwift/ViewController.swift:1719]: can we remove the printing of the eth balance in the logs

File: Sources/PortalSwift/Portal.swift [hunk 2]
```diff
@@ -101,7 +101,7 @@ public class Portal {
 
     // Creating this as a variable first so it's usable to
     // fetch the client in the Task at the end of the initializer
-    let api = api ?? PortalApi(apiKey: apiKey, apiHost: apiHost, provider: provider)
+    let api = api ?? PortalApi(apiKey: apiKey, apiHost: apiHost, provider: self.provider)
```
Comment (Rshahatit) [inline Sources/PortalSwift/Portal.swift:104]: i think you dont have the correct swift format settings. Can you ask blake about how to get that set up properly for your development environment.

Comment (Rshahatit) [reply Sources/PortalSwift/Portal.swift:104]: thats why its adding a self everywhere.

File: Sources/PortalSwift/Portal.swift [hunk 3]
```diff
@@ -349,6 +355,30 @@ public class Portal {
     )
   }
 
+  /// You should only call this function if you are upgrading from v3 to v4.
```
Comment (Rshahatit) [inline Sources/PortalSwift/Portal.swift:358]: is it possible to make this comment a decorator that pops up like a deprecated symbol or not?

**Review by Rshahatit — APPROVED** [review]
LGTM

File: Sources/PortalSwift/Portal.swift (hunk 3 above)
Comment (AhmedRagabIssa) [reply Sources/PortalSwift/Portal.swift:358]: Screenshot 2024-07-31 at 4 34 37 AM

Adding a comment with `///` 3 forward slashes is adding documentation to the developer appears with the Xcode auto-complete by default.

**Pull request merged**